onnxscript/ir/_core.py

```diff
@@ -1766,6 +1764,8 @@ def num_nodes(self) -> int:
     def append(self, node: Node, /) -> None:
         """Append a node to the graph in O(1) time.
 
+        Unique names will be assigned to the node and its values if any name is ``None``.
```

Comment:
Are these the only places where the naming happens? When a node is added into a graph? Just trying to get the big picture. It makes sense.

Reply:
Does deserialization of a proto also call these node-insertion methods when creating the graph?

Reply:
> Does deserialization of a proto also call these node-insertion methods when creating the graph?

Yes they do call

Reply:
> Are these the only places where the naming happens? When a node is added into a graph? Just trying to get the big picture. It makes sense.

Yes, it only happens when a node is added into a graph currently